replace public references to twitter/X with Matrix

jmuzina opened the pull request:
## Done

[List of work items including drive-bys]

Fixes [list issues/bugs if needed]

## QA

- Open [demo](insert-demo-url)
- [Add QA steps]
- Review updated documentation:
  - [List any updated documentation for review]

### Check if PR is ready for release

If this PR contains Vanilla SCSS or macro code changes, it should contain the following changes to make sure it's ready for the release:

- [ ] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [ ] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical/vanilla-framework/releases/latest), following semver convention
  - if existing APIs (CSS classes & macro APIs) are not changed it can be a bugfix release (x.x.**X**)
  - if existing APIs (CSS classes & macro APIs) are changed/added/removed it should be a minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [ ] Any changes to component class names (new patterns, variants, removed or added features) or macros should be listed on the [what's new page](https://github.com/canonical/vanilla-framework/blob/main/releases.yml).


## Screenshots

[if relevant, include a screenshot or screen capture]

webteam-app commented:
### [<img src='https://assets.ubuntu.com/v1/6baef514-ubuntu-circle-of-friends-large.svg' height=32 width=32> Demo</img>](https://vanilla-framework-5625.demos.haus)
### [<img src='https://assets.ubuntu.com/v1/e512b0e2-jenkins.svg' height=32 width=32> Jenkins </img>](https://jenkins.canonical.com/webteam/job/start-demo/18551)
### [<img src='https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg' height=32 width=32> demos.haus </img>](https://demos.haus)

juanruitina reviewed (changes requested):
We should also update the links in templates/docs/index.html (link to Slack, "Tweet us"...).

I'd also update any references in the examples to Twitter (remove or refer to X, for instance in icons) and Slack (replacing with Matrix). But I understand if you'd rather handle this separately.

jmuzina commented:
@juanruitina Thanks for your points!

> We should also update the links in templates/docs/index.html (link to Slack, "Tweet us"...)

Good point, I've updated these to replace the link for "Join our channel" with our Matrix link, and removed the twitter link.

<img width="1013" height="594" alt="image" src="https://github.com/user-attachments/assets/f23a1ae8-5b11-4574-b737-514e3182de6a" />

> I'd also update any references in the examples to Twitter (remove or refer to X, for instance in icons) and Slack (replacing with Matrix).

I've removed twitter linkx from examples that were merely a text link, but kept the X icon link in the [icon links example](https://vanilla-framework-5625.demos.haus/docs/examples/patterns/icons/icons-links?theme=light) to keep a working example of using an X icon and maintain visual testing coverage. I've updated the link address there to point to the [Canonical X page](https://x.com/Canonical) instead of the Vanilla page. 

Visual diffs can be seen on [Percy](https://percy.io/bb49709b/web/vanilla-framework/builds/42577921/changed/2271616467?browser=chrome&browser_ids=69&group_snapshots_by=similar_diff&subcategories=unreviewed%2Cchanges_requested&utm_source=github_status_private&viewLayout=overlay&viewMode=new&width=375&widths=375%2C800%2C1280)

juanruitina commented:
I think ""Join us on Matrix" or "Join the discussion on Matrix" works better than "Join the channel"; Matrix doesn't use "channels" but "rooms". All the rest looks good to me :)
